README.md

```diff
@@ -105,6 +114,88 @@ registry.TryRegister<Index>(indexer); // register for DI
 registry.Get<Index>(); // use in DI
 ```
 
+### Injecting Actors with `IRequiredActor<TKey>`
+
+Suppose we have a class that depends on having a reference to a top-level actor, a router, a `ShardRegion`, or perhaps a `ClusterSingleton` (common types of actors that often interface with non-Akka.NET parts of a .NET application):
+
+```csharp
+public sealed class MyConsumer
+{
+    private readonly IActorRef _actor;
+
+    public MyConsumer(IRequiredActor<MyActorType> actor)
+    {
+        _actor = actor.ActorRef;
+    }
+
+    public async Task<string> Say(string word)
+    {
+        return await _actor.Ask<string>(word, TimeSpan.FromSeconds(3));
+    }
+}
+```
+
+The `IRequiredActor<MyActorType>` will cause the Microsoft.Extensions.DependencyInjection mechanism to resolve `MyActorType` from the `ActorRegistry` and inject it into the `IRequired<Actor<MyActorType>` instance passed into `MyConsumer`.
+
+The `IRequiredActor<TActor>` exposes a single property:
+
+```csharp
+public interface IRequiredActor<TActor>
+{
+    /// <summary>
+    /// The underlying actor resolved via <see cref="ActorRegistry"/> using the given <see cref="TActor"/> key.
+    /// </summary>
+    IActorRef ActorRef { get; }
+}
+```
+
+By default, you can automatically resolve any actors registered with the `ActorRegistry` without having to declare anything special on your `IServiceCollection`:
+
+```csharp
+using var host = new HostBuilder()
+  .ConfigureServices(services =>
+  {
+      services.AddAkka("MySys", (builder, provider) =>
+      {
+          builder.WithActors((system, registry) =>
+          {
+              var actor = system.ActorOf(Props.Create(() => new MyActorType()), "myactor");
+              registry.Register<MyActorType>(actor);
+          });
+      });
+      services.AddScoped<MyConsumer>();
+  })
+  .Build();
+  await host.StartAsync();
+```
+
+Adding your actor and your type key into the `ActorRegistry` is sufficient - no additional DI registration is required to access the `IRequiredActor<TActor>` for that type.
+
+#### Resolving `IRequiredActor<TKey>` within Akka.NET
+
+Akka.NET does not use dependency injection to start actors by default primarily because actor lifetime is unbounded by default - this means reasoning about the scope of injected dependencies isn't trivial. ASP.NET, by contrast, is trivial: all HTTP requests are request-scoped and all web socket connections are connection-scoped - these are objects have _bounded_ and typically short lifetimes.
+
+Therefore, users have to explicitly signal when they want to use Microsoft.Extensions.DependencyInjection via [the `IDependencyResolver` interface in Akka.DependencyInjection](https://getakka.net/articles/actors/dependency-injection.html) - which is easy to do in most of the Akka.Hosting APIs for starting actors:
+
+```csharp
+protected override void ConfigureServices(HostBuilderContext context, IServiceCollection services)
```

Comment:
This sample is a bit confusing because it is using `Akka.Hosting.TestKit` syntax, users would not use this code.